### PR TITLE
enh(ci): refactor get-version workflow

### PR DESCRIPTION
**Pull request by kduret**
## Description

enh(ci): refactor get-version workflow

* use nodejs instead of bash (80%)
* remove usage of php to parse version file
* detect properly cloud versions
* manage feature branches
* generate summary
* allow usage of of `database` and `system` PR labels to run according e2e tests
* switch public runners to `ubuntu-24.04`
 
image

**Fixes** MON-150464

### CODE REVIEW

File: .github/workflows/get-environment.yml [hunk 1]
```diff
@@ -0,0 +1,321 @@
+on:
+  workflow_call:
+    inputs:
+      version_file:
+        required: true
+        type: string
+    outputs:
+      latest_major_version:
+        description: "latest major version"
+        value: ${{ jobs.get-environment.outputs.major_version }}
```
Comment (mushroomempires) [inline .github/workflows/get-environment.yml:10]:
```suggestion
        value: ${{ jobs.get-environment.outputs.latest_major_version }}
```

File: .github/workflows/get-environment.yml [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+            RELEASE="1"
+          else
+            RELEASE="$(date +%s).$(echo ${{ github.sha }} | cut -c -7)"
+          fi
+
+          echo "release=$RELEASE" >> $GITHUB_OUTPUT
+        shell: bash
+
+      - name: "Get release type: hotfix, release or not defined if not a release"
+        id: get_release_type
```
Comment (mushroomempires) [inline .github/workflows/get-environment.yml:196]: is it really "safe" to not define the release type in every case here?

File: .github/workflows/test-get-environment.yml
Comment (mushroomempires) [inline .github/workflows/test-get-environment.yml:1]: probably isn't necessary anymore

**Review by mushroomempires — CHANGES_REQUESTED** [review]

File: .github/workflows/get-environment.yml (hunk 2 above)
Comment (kduret) [reply .github/workflows/get-environment.yml:196]: Yes if value is properly surrounded by quotes in bash scripts

**Review by mushroomempires — DISMISSED** [review]

File: .github/docker/centreon-web/alma8/Dockerfile.dependencies [hunk 3]
```diff
@@ -21,7 +21,7 @@ dnf config-manager --set-enabled powertools
 
 dnf install -y epel-release
 
-if [[ ${RELEASE_CLOUD} -eq 1 ]]; then
+if [[ ${IS_CLOUD} == "true" ]]; then
```
Comment (kduret) [inline .github/docker/centreon-web/alma8/Dockerfile.dependencies:24]:
```suggestion
if [[ "${IS_CLOUD}" == "true" ]]; then
```

File: .github/docker/centreon-web/alma9/Dockerfile.dependencies [hunk 4]
```diff
@@ -21,7 +21,7 @@ dnf config-manager --set-enabled crb
 
 dnf install -y epel-release
 
-if [[ ${RELEASE_CLOUD} -eq 1 ]]; then
+if [[ ${IS_CLOUD} == "true" ]]; then
```
Comment (kduret) [inline .github/docker/centreon-web/alma9/Dockerfile.dependencies:24]:
```suggestion
if [[ "${IS_CLOUD}" == "true" ]]; then
```

**Review by mushroomempires — APPROVED** [review]

**Review by tuntoja — APPROVED** [review]

**Pull request merged**